From 56f3b64a6f3b4b77e0c185d460806c483e302b96 Mon Sep 17 00:00:00 2001
From: Ayyappa Ch <ayyappa.chandolu@amd.com>
Date: Mon, 6 Feb 2017 19:00:52 +0530
Subject: [PATCH 1716/1722] drm/amdgpu: Don't clear BO during memory allocation

vector_copy on dgpu hangs due to clearing the bo at
allocation.

Signed-off-by: Ayyappa Ch <ayyappa.chandolu@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c | 16 ----------------
 1 file changed, 16 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c
index 3eef2be..e877221 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c
@@ -613,22 +613,6 @@ static int __alloc_memory_of_gpu(struct kgd_dev *kgd, uint64_t va,
 	bo->kfd_bo = *mem;
 	(*mem)->bo = bo;
 
-	if (domain == AMDGPU_GEM_DOMAIN_VRAM) {
-		ret = amdgpu_bo_reserve(bo, true);
-		if (ret) {
-			dev_err(adev->dev,
-				"(%d) failed to reserve bo for amdkfd\n", ret);
-			goto err_bo_clear;
-		}
-		ret = amdgpu_amdkfd_gpuvm_clear_bo(adev, vm, bo);
-		amdgpu_bo_unreserve(bo);
-		if (ret) {
-			pr_err("amdkfd: failed to clear VRAM BO object. ret %d\n",
-					ret);
-			goto err_bo_clear;
-		}
-	}
-
 	pr_debug("amdkfd: created BO on domain %d with size %llu\n",
 				alloc_domain, size);
 
-- 
2.7.4

